src/js/components/ContainerSettingsComponent.jsx

```diff
@@ -14,11 +14,28 @@ var ContainerSettingsComponent = React.createClass({
   },
 
   getInitialState: function () {
+    var container = this.props.model.container;
+
+    var [portMappingsCount, parametersCount, volumesCount] = Array(3).fill(1);
```

Comment:
Actually, why not have a configuration map? That way you can loop through it and make this thing much more dynamic. Say we wanted to add another field.

```
var config = {
  portMappings: {
    count: 1,
    getProperty: (obj) => obj.docker.portMappings.length || 1;
  },
  parameters: {
    count: 1,
    getProperty: (obj) => obj.parameters.length || 1;
  },
  volumes: {
    count: 1,
    getProperty: (obj) => obj.volumes.length || 1;
  }
};

if (container != null) {
  Object.keys(config).forEach(function (key) {
    var field = config[key];
    field.count = field.getProperty(container);
  });
}

// do something with rows as well
```

Or something similar. This way, we can just add to the configuration